modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java

```diff
@@ -300,12 +301,24 @@ private void loadConfiguration(boolean evalDynamicProperties) throws Configurati
      * @since 5.7
      */
     protected void includeTemplates() throws ConfigurationException {
+        // include nuxeo.templates
         String templates = configHolder.getProperty(PARAM_TEMPLATES_NAME);
         if (isBlank(templates)) {
             log.warn("No template found in configuration! Fallback on 'default'.");
             templates = "default";
             configHolder.put(PARAM_TEMPLATES_NAME, templates);
         }
+        // include nuxeo.templates.*
+        String templatesWildcard = configHolder.stringPropertyNames()
+                                               .stream()
+                                               .filter(k -> k.startsWith(PARAM_TEMPLATES_NAME + '.'))
+                                               .sorted()
+                                               .map(configHolder::getProperty)
+                                               .collect(Collectors.joining(TEMPLATE_SEPARATOR));
+        if (isNotBlank(templatesWildcard)) {
+            templates += TEMPLATE_SEPARATOR + templatesWildcard;
+        }
```

Comment:
I'm continuing the discussion here in order to have a thread, @jcarsique:
> Is there no FreeMarker overlap between "nuxeo.templates" and "nuxeo.templates.*" ?

It seems there is a freemarker WARN log when both properties exist, see below:

```lang-none
WARN  [TextTemplate] FreeMarker variables: ignored 'nuxeo.templates.cloud' conflicting with 'nuxeo.templates'
```

If I understand correctly, it means that `nuxeo.templates.cloud` won't be available for variable substitution in freemarker templating.

I don't think it is something we want, but I may be wrong.

As it is our code, and depending on what we want for these variables, do you think we can ignore the log in `TextTemplate`?

Reply:
Yes: FreeMarker does not like its variables to overlap ("a.b" vs "a.b.c") and that makes it buggy if used in a substitution.
I would rather try to migrate and remove the default "nuxeo.templates" to "nuxeo.templates.all" or anything similar.
Or use "nuxeo.template.new" instead of "nuxeo.templates.new".

Reply:
Confirmed again today: that will make `nuxeo.templates` being a Map instead of String, thus breaking any current usage in the templates.

Reply:
Ok, it doesn't like a way to go.

Migrating the current to `nuxeo.templates.all` seems to me a huge breaking change, even if I break everything regarding java code, we should be good regarding _nuxeo.conf_.

I'll check with the team what we can do. 

Thanks for the details.

Reply:
We've decided to move from `nuxeo.templates.*` to `nuxeo.append.templates.*` to avoid this issue.

I'm resolving this thread, don't hesitate to re-open it if needed.